src/sif/autocost.cc

```diff
@@ -285,6 +282,28 @@ class AutoCost : public DynamicCost {
     return static_cast<uint8_t>(type_);
   }
 
+  bool IsHOVAllowed(const baldr::DirectedEdge* edge) const {
+    // A non-hov edge means hov is allowed.
+    if (!edge->is_hov_only())
+      return true;
```

Comment:
this is not inlined and its in the tight loop. i suspect that there is a performance impact from this. the good news is that this first `if` is almost always hit so we dont have to hit 3 more branches below. still it would be nice to at least test performacne difference. maybe you could do it as a follow up? like we can merge the pr but you can post back your findings regarding any changes in performance?

Reply:
@kevinkreiser I ran some benchmarks this morning (`make run-benchmarks` and `valhalla_run_route`).  I did a bunch of runs, did some averaging... I don't think I'm measuring a difference.  That said, I struggle to get stable perf timings  on my system (mac laptop).